### PR TITLE
Es6 projects common modules email run checks

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

Such ES6, much Flow

## What is the value of this and can you measure success?

Everything is better with ES6

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/email/run-checks.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        response &&
+        response.status !== 'error' &&
+        response.result &&
+        response.result.subscriptions
+    ) {
+        userListSubs = response.result.subscriptions.map(sub => sub.listId);
+        userListSubsChecked = true;
+    }
+
+    return userListSubs;
```
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/email/run-checks.js]: :woman_facepalming:

File: static/src/javascripts/projects/common/modules/email/run-checks.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    !!$el.nodeName && $el.nodeName === 'P';
+
+const allowedArticleStructure = (): boolean => {
+    const $articleBody = $('.js-article__body');
+
+    if ($articleBody.length) {
+        const allArticleEls = $('> *', $articleBody);
+
+        return [].slice
+            .call(allArticleEls, allArticleEls.length - 2)
```
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/email/run-checks.js]:
```js
allArticleEls.slice(allArticleEls.length - 2)
```

?

File: static/src/javascripts/projects/common/modules/email/run-checks.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+};
+
+const getEmailShown = (): string => emailShown;
+
+const allEmailCanRun = (): boolean =>
+    !config.get('page.shouldHideAdverts') &&
+    !config.get('page.isSensitive') &&
+    !config.get('page.isFront') &&
+    (config.get('page.contentId') &&
+        config.get('page.contentId').indexOf('email-sign-up') === -1) &&
```
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/email/run-checks.js]:
```js
!config.get('page.contentId').includes('email-sign-up')
```

**Review by gu-stav — APPROVED** [review]

File: static/src/javascripts/projects/common/modules/email/run-checks.js (hunk 2 above)
Comment (GHaberis) [reply static/src/javascripts/projects/common/modules/email/run-checks.js]: If `$` returns a `bonzo` can you call `slice` directly on a bonzo list? 

You could try `[...allArticleEls].slice(allArticleEls.length - 2)`

**Review by GHaberis — APPROVED** [review]

File: static/src/javascripts/projects/common/modules/email/run-checks.js (hunk 2 above)
Comment (SiAdcock) [reply static/src/javascripts/projects/common/modules/email/run-checks.js]: The bad news: neither of those patterns works for `bonzo` 🤖 
The good news `Array.from(...)` does work 🤠

**Pull request merged**